### PR TITLE
'istioctl verify-install' consume IstioOperator

### PR DESCRIPTION
**Pull request by esnible**
Part One of https://github.com/istio/istio/issues/18547

If an IstioOperator is passed to `istioctl verify install -f`, the logic here will generate a manifest from the IstioOperator and using the existing logic to verify that all of the manifest items are present in the cluster.

(This lacks tests.  The current code never had any tests.  Adding testing would be difficult -- it would mean mocking an entire Istio install via Kube API.  Thoughts?

A follow-up will allow verify-install to consume the IstioOperator preserved in the cluster, to avoid the need to specify a `-f` option.

cc @irisdingbj

### CODE REVIEW

File: istioctl/pkg/install/verify.go [hunk 1]
```diff
@@ -122,10 +151,46 @@ func verifyPostInstall(enableVerbose bool, istioNamespaceFlag string,
 			for _, c := range job.Status.Conditions {
 				if c.Type == v1batch.JobFailed {
 					msg := fmt.Sprintf("Istio installation failed, incomplete or"+
-						" does not match \"%s\" - the required Job  %s failed", options.Filenames[0], name)
+						" does not match \"%s\" - the required Job %s failed", filename, name)
 					return errors.New(msg)
 				}
 			}
+		case "IstioOperator":
+			result := info.Client.
```
Comment (irisdingbj) [inline istioctl/pkg/install/verify.go:159]: Does this mean we will always use the `IstioOperator` stored in cluster other than the one specified via -f ? 
I think we should do like: 
if no `-f` specified, we will perform post install verification using the IstioOperator stored in cluster.
if `-f` specified, we will use it directly to generate manifest and do verify.

**Comment by irisdingbj** [comment]
@esnible Thanks for this! 
Other question is do we want to support `--set` command line option similar with `manifest apply`? eg: `istioctl verify-install --set values.global.xx=xxx`?

**Comment by esnible** [comment]
> @esnible Thanks for this!
> Other question is do we want to support `--set` command line option similar with `manifest apply`? eg: `istioctl verify-install --set values.global.xx=xxx`?

I thought about it.  Now that I have added support for `--revision` it should be rare that anyone uses `-f <iop.yaml>`.  If there is demand we can add this later.

**Comment by esnible** [comment]
@ostromart I have added code to use the IstioOperator your PR places into the cluster.  I would be curious to have you review the `operatorFromCluster()` method.  I really struggled with it ... not sure why the default REST handling doesn't know about IstioOperator and I couldn't find a way to even get the logic to treat it as an Unstructured.

File: istioctl/pkg/install/verify.go [hunk 2]
```diff
@@ -270,3 +356,122 @@ func findResourceInSpec(kind string) string {
...
+	// TODO use result.Object() insted of explicitly unmarshaling into Unstructured
+	raw, err := result.Raw()
+	if err != nil {
+		return nil, err
+	}
+	var un unstructured.Unstructured
+	err = json.Unmarshal(raw, &un)
+	if err != nil {
+		return nil, err
+	}
```
Comment (richardwxn) [inline istioctl/pkg/install/verify.go:436]: why not just directly define gvr scheme and use dynamic client to get unstructure list

Comment (esnible) [reply istioctl/pkg/install/verify.go:436]: > why not just directly define gvr scheme and use dynamic client to get unstructure list

I am willing to do that.  I can use the IstioOperatorGVK in https://godoc.org/istio.io/istio/operator/pkg/apis/istio/v1alpha1#pkg-variables with the dynamic client.

I thought was bad form to use two different clients in the same subcommand, but perhaps not.  I will try it out this evening.

Comment (esnible) [reply istioctl/pkg/install/verify.go]: Thanks @richardwxn !  Using the dynamic client simplified the lookup considerably.

**Pull request merged**